### DISCUSSION TITLE
Tear down everything on uninstall?

### DISCUSSION THREAD

**Discussion by tuxillo** [fluxcd/flux2#3815] [General] [answered]
When doing a `fluxcd uninstall`, all the reconciled items are left untouched, I guess this is to prevent accidents. But is there any automated way to tear down everything that was put in place by FluxCD without doing it manually?

My particular case is that on a AWS EKS cluster created by Terrraform, the nginx ingress LoadBalancer service that is defined in my FluxCD setup creates a load balancer in AWS that has to be deallocated before `terraform destroy` or otherwise the destruction won't be completed due to dependencies.

**Comment by makkes** [comment]
https://fluxcd.io/flux/components/kustomize/kustomization/#prune

**Comment by stefanprodan** [comment, marked as answer]
Tear down of the reconciled workloads is not supported and is out of scope for Flux CLI.

You could write your own logic that deletes PVC, LoaldBalancers, Namespaces in the right order after running `flux uninstall`. Each cloud behaves differently at tear down and it’s not feasible for Flux to deal with all these variation, for example to remove AWS AppMesh from EKS requires a bunch of calls to the AWS API with custom wait/retry logic between them otherwise the cluster can’t be deleted, and this is just one of many edge cases like this.

**Reply by tuxillo** [reply]
Thanks.
